### PR TITLE
fix: improve versioning documentation and add showcase example

### LINKED ISSUE #162: improve versioning documentation and add an example to showcase it
# Description of the issue

Improve versioning documentation.

I update the versioning documentation and i add an example to showcase it.

## Reason

Without going into too much detail this was blocking me to test that there was no issues when changing to ESM due to the fact that we need to make changes to the loading of the resolvers and this impacts the versioning method.

I did lose a ton of time due to a mistake while following the documentation and i decided to improve it slightly.

### PR DESCRIPTION
**Pull request by Xavier-Redondo**
# Pull Request Template

## PR Checklist

- [x] I have run `npm test` locally and all tests are passing.
- [ ] I have added/updated tests for any new behavior.
- [x] I have added/updated documentation for any new behavior.
- [x] If this is a significant change, an issue has already been created where the problem / solution was discussed: [N/A, or add link to issue here]

## PR Description

No functional changes at all
This pr has two parts:

- creation of a new example project that showcases how the api versioning works. I need this in order to make sure that any change applied due to ESM update does not break the versioning mechanism.
- improve the current documentation to clearly specify what is required to make it work (previous documentation had an example without deprecated,  and it did not clearly specify that deprecated was mandatory). I have slightly updated the examples to have them work.

Out of scope of this PR: Before there was no example with express and there is still no example of express. In fact i have removed the reference to it and a TBD due to our plans with it.

This closes #162

### CODE REVIEW

File: packages/bautajs-core/src/bauta.ts [hunk 1]
```diff
@@ -274,8 +275,10 @@ export class BautaJS implements BautaJSInstance {
         'The given instance is not bootstrapped, thus operation schema will be no inherited.'
       );
     }
+
```
Comment (franher) [inline packages/bautajs-core/src/bauta.ts:278]: @Xavier-Redondo please do not commit the changes on this file.

File: packages/bautajs-fastify/src/index.ts [hunk 2]
```diff
@@ -26,10 +26,12 @@ export async function bautajsFastify(
   fastify: FastifyInstance,
   opts: BautaJSFastifyPluginOptions
 ): Promise<void>;
+
```
Comment (franher) [inline packages/bautajs-fastify/src/index.ts:29]: @Xavier-Redondo do not commit this file please

**Review by franher — CHANGES_REQUESTED** [review]
LGTM, but please do not commit the changes outside the bautajs-fastify-versioning-example.

Thank you for your contribution :)

File: packages/bautajs-core/src/bauta.ts (hunk 1 above)
Comment (Xavier-Redondo) [reply packages/bautajs-core/src/bauta.ts]: done

File: packages/bautajs-fastify/src/index.ts (hunk 2 above)
Comment (Xavier-Redondo) [reply packages/bautajs-fastify/src/index.ts]: done

**Pull request merged**